VAR-SOM-AM62P: JIG initial support

FrancescoFerraro opened the pull request.

varigigi reviewed (approved):
@FrancescoFerraro, the changes look fine to me.
As usual for JIG, please wait for @aviadh80 / @eranmati review too.

aviadh80 commented on arch/arm64/boot/dts/ti/k3-am62p5-var-som-jig.dts:
```diff
@@ -0,0 +1,516 @@
+		};
+	};
+
+	gpio-leds {
```
The gpio-leds node should be removed same as in k3-am625-var-som-jig.dts

aviadh80 commented on arch/arm64/boot/dts/ti/k3-am62p5-var-som-jig.dts:
```diff
@@ -0,0 +1,516 @@
+&main_mcan0 {
+	status = "disabled";
+};
+
```
Not sure what the current maximum GPIOs number it set to in the AM62P , however for AM62 the setting was less than the maximum 92 GPIO which are possible.
In order to have all accessible, we added this node in k3-am625-var-som-jig.dts

&main_gpio0 {
        ti,ngpio = <92>;
};

FrancescoFerraro replied:
Hi @aviadh80 
ti,ngpio is already 92 from TI: https://github.com/varigit-dev/ti-linux-kernel/blob/0446b9df48ed43dee08ec34491a7062e9a555b5f/arch/arm64/boot/dts/ti/k3-am62p-main.dtsi#L62

aviadh80 commented on arch/arm64/boot/dts/ti/k3-am62p5-var-som-jig.dts:
```diff
@@ -0,0 +1,516 @@
+
+		clocks = <&clk_ov5640_fixed>;
+		clock-names = "xclk";
+
```
these GPIOs are not connected in the JIG adaptor, powerdown-gpios,reset-gpios should be removed same as in k3-am625-var-som-jig.dts

aviadh80 commented on arch/arm64/boot/dts/ti/k3-am62p5-var-som-jig.dts:
```diff
@@ -0,0 +1,516 @@
+		>;
+		bootph-all;
+	};
+
```
GPIO0_53 is used by vdd_mmc1_int: regulator-2 so need to add the below same as k3-am62p5-var-som-symphony.dts just without GPIO 0_30 and also add it to :

```
vdd_mmc1_pins_default: vdd-mmc1-pins-default {
		pinctrl-single,pins = <
			
			AM62PX_IOPAD(0x00d8, PIN_OUTPUT, 7) /* (AE22) VOUT0_DATA8.GPIO0_53 */
		>;
		bootph-all;
	};
```

aviadh80 commented on arch/arm64/boot/dts/ti/k3-am62p5-var-som-jig.dts:
```diff
@@ -0,0 +1,516 @@
+	vdd_mmc1_int: regulator-2 {
+		compatible = "regulator-fixed";
+		regulator-name = "sd_int_rst";
+		regulator-min-microvolt = <3300000>;
```
need to add pinctrl for GPIO0_53 same as in k3-am62p5-var-som-symphony.dts

aviadh80 reviewed (commented):
@FrancescoFerraro @eranmati see my comments inline

FrancescoFerraro committed:
am62px-var-som: add initial JIG support for VAR-SOM-AM62P

Signed-off-by: FrancescoFerraro <francesco.f@variscite.com>

FrancescoFerraro commented:
@aviadh80 I made the changes following your review.
Thanks

leonid-s committed:
VAR-SOM-AM62P Rev 1.1: Initial support added

leonid-s committed:
am62px-var-som: First working JIG version

 Signed-off-by: Leonid Segal <leonid.s@variscite.com>

leonid-s committed:
am62px-var-som: ti,min-output-impedance field added to the PHY definition

Signed-off-by: Leonid Segal <leonid.s@variscite.com>

yairbenavraham committed:
arm64: dts: ti: add AM62P jig GPIO expanders

Add the PCA9534 GPIO expanders at I2C addresses 0x21 and 0x24 on the
VAR-SOM-AM62P JIG board.

Signed-off-by: Yair Ben Avraham <yair.b@variscite.com>
